app/src/main/java/com/github/multimatum_team/multimatum/ProfileActivity.kt

```diff
@@ -38,19 +40,23 @@ class ProfileActivity : AppCompatActivity() {
 
         // Update the e-mail when the account changes.
         userViewModel.getUser().observe(this) { user ->
-            updateUI(user)
+            if (user is SignedInUser) {
+                updateUI(user)
+            } else {
+                throw IllegalStateException("ProfileActivity launched while the current user is not signed in")
+            }
         }
     }
 
     /**
      * Update the e-mail with the new user.
      */
-    private fun updateUI(user: User) {
+    private fun updateUI(user: SignedInUser) {
         loginMessage.text =
-            getString(R.string.you_re_logged_in_as, (user as SignedInUser).email)
+            getString(R.string.you_re_logged_in_as, user.email)
     }
 
     companion object {
-        private const val TAG = "ProfileFragment"
+        private const val TAG = "ProfileActivity"
```

Comment:
Is that for logging? That is just a suggestion, but you could maybe use the functions from `LogUtil` to make logging more practical